R/find-port.R

```diff
@@ -44,5 +44,5 @@ findPort <- function(port){
     stop("Unable to start a Plumber server. Either the port specified was unavailable or we were unable to find a free port.")
   }
 
-  port
+  as.integer(port)
```

Comment:
The problem is that this doesn't fully handle validation of user input because all of the code above it assumes that `port` wasn't provided.

For example, given obviously bad user input both doesn't work AND doesn't error.

```r
options(plumber.port = "blue"); plumber::pr_run(plumber::pr())
#> Running plumber API at http://127.0.0.1:NA
#> Running swagger Docs at http://127.0.0.1:NA/__docs__/
```

Reply:
I refactored `findPort()` so that, in the missing port case we immediately hand over control to `findRandomPort()` which handles finding a random port.

This leaves the rest of `findPort()` to handle coercing user input to an integer and checking that it's a valid value.

Reply:
Right, Thanks. It seems I only addressed the narrower issue that (correct) ports specified via env var were failing. Didn't expect `NA` to be accepted as a port, should this not fail somewhere else? What happens if an out of port range integer is specified?

Reply:
Thanks @shikokuchuo, and yeah somehow `NA` would flow through because `findPort()` didn't currently do any validating of the port number. You're right, we should also check the valid port range, that wouldn't fail later either. I'll add that to this PR.